pkg/backend/login_manager.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2016-2024, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package backend
+
+import (
+	"context"
+
+	pkgWorkspace "github.com/pulumi/pulumi/pkg/v3/workspace"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/diag"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/diag/colors"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/workspace"
+)
+
+// LoginManager provides a slim wrapper around functions related to backend logins.
+type LoginManager interface {
+	// Current returns the currently logged in backend instance for the given url.
+	Current(
+		ctx context.Context,
+		ws pkgWorkspace.Context,
+		sink diag.Sink,
+		url string,
+		project *workspace.Project,
+		setCurrent bool,
```

Comment:
We currently _always_ set this to true and I was going to remove it but it looks like AI team want to start using this without setting current https://github.com/pulumi/pulumi/pull/16825/files#diff-3757d38f874f2973612507fde4e607d757d54b8caa9525b433d04769d794f5c9R182

Reply:
Thank you! 🙏